bridges/snowbridge/parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -0,0 +1,342 @@
+				} else {
+					*nonce = nonce.saturating_add(1);
+					Ok(())
+				}
+			})?;
+
+			// Reward relayer from the sovereign account of the destination parachain
+			// Expected to fail if sovereign account has no funds
+			let sovereign_account = sibling_sovereign_account::<T>(channel.para_id);
+			let delivery_cost = Self::calculate_delivery_cost(message.encode().len() as u32);
+			T::Token::transfer(&sovereign_account, &who, delivery_cost, Preservation::Preserve)?;
+
+			// Decode message into XCM
+			let (xcm, fee) =
+				match inbound::VersionedMessage::decode_all(&mut envelope.payload.as_ref()) {
+					Ok(message) => Self::do_convert(envelope.message_id, message)?,
+					Err(_) => return Err(Error::<T>::InvalidPayload.into()),
+				};
+
+			// We embed fees for xcm execution inside the xcm program using teleports
+			// so we must burn the amount of the fee embedded into the XCM script.
+			T::Token::burn_from(&sovereign_account, fee, Precision::Exact, Fortitude::Polite)?;
+
+			log::info!(
+				target: LOG_TARGET,
+				"💫 xcm {:?} sent with fee {:?}",
+				xcm,
+				fee
+			);
+
+			// Attempt to send XCM to a dest parachain
+			let message_id = Self::send_xcm(xcm, channel.para_id)?;
+
+			Self::deposit_event(Event::MessageReceived {
+				channel_id: envelope.channel_id,
+				nonce: envelope.nonce,
+				message_id,
+			});
+
+			Ok(())
+		}
+
+		/// Halt or resume all pallet operations. May only be called by root.
+		#[pallet::call_index(1)]
+		#[pallet::weight((T::DbWeight::get().reads_writes(1, 1), DispatchClass::Operational))]
+		pub fn set_operating_mode(
+			origin: OriginFor<T>,
+			mode: BasicOperatingMode,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+			OperatingMode::<T>::set(mode);
+			Self::deposit_event(Event::OperatingModeChanged { mode });
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn do_convert(
+			message_id: H256,
+			message: inbound::VersionedMessage,
+		) -> Result<(Xcm<()>, BalanceOf<T>), Error<T>> {
+			let (mut xcm, fee) =
+				T::MessageConverter::convert(message).map_err(|e| Error::<T>::ConvertMessage(e))?;
+			// Append the message id as an XCM topic
+			xcm.inner_mut().extend(vec![SetTopic(message_id.into())]);
+			Ok((xcm, fee))
+		}
+
+		pub fn send_xcm(xcm: Xcm<()>, dest: ParaId) -> Result<XcmHash, Error<T>> {
+			let dest = MultiLocation { parents: 1, interior: X1(Parachain(dest.into())) };
+			let (xcm_hash, _) = send_xcm::<T::XcmSender>(dest, xcm).map_err(Error::<T>::from)?;
+			Ok(xcm_hash)
+		}
+
+		pub fn calculate_delivery_cost(length: u32) -> BalanceOf<T> {
+			let weight_fee = T::WeightToFee::weight_to_fee(&T::WeightInfo::submit());
+			let len_fee = T::LengthToFee::weight_to_fee(&Weight::from_parts(length as u64, 0));
+			weight_fee
+				.saturating_add(len_fee)
+				.saturating_add(T::PricingParameters::get().rewards.local)
+		}
+	}
+
+	/// API for accessing the delivery cost of a message
+	impl<T: Config> Get<BalanceOf<T>> for Pallet<T> {
```

Comment:
IIUC this `Get<>` implementation is no longer required after https://github.com/Snowfork/snowbridge/pull/1065?

Reply:
Thanks for the approve. We still use it to set the delivery price for messages on the ethereum side:
https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/system/src/lib.rs#L332